oak_functions/lookup/src/lib.rs

```diff
@@ -99,6 +99,37 @@ impl<L: OakLogger> OakApiNativeExtension for LookupData<L> {
 // value](https://github.com/project-oak/oak/tree/main/oak_functions/lookup/README.md#invariant-at-most-one-value)
 pub type Data = HashMap<Vec<u8>, Vec<u8>>;
 
+enum BuilderState {
+    New,
+    Updating,
+}
+
+// Incrementally build data backing the lookup data keeping track of the state.
+struct DataBuilder {
+    data: Data,
+    state: BuilderState,
+}
+
+impl DataBuilder {
+    fn new() -> Self {
+        DataBuilder {
+            data: Data::new(),
+            state: BuilderState::New,
+        }
+    }
+
+    fn build(&mut self) -> Data {
+        self.state = BuilderState::New;
+        core::mem::replace(&mut self.data, Data::new())
+    }
+
+    // Assumes data and new data and data are disjoint.
```

Comment:
nit: for consistency this comment should be inside the function, or should be a sub-point on a doc comment that describes what the function does, or could just be removed since new values will just overwrite old values if the keys are not disjoint.